src/common/src/types/mod.rs

```diff
@@ -447,6 +490,10 @@ impl From<DataType> for PbDataType {
 mod private {
     use super::*;
 
+    // Note: put pub trait inside a private mod just makes the name private,
+    // The trait methods will still be publicly available...
+    // a.k.a. ["Voldemort type"](https://rust-lang.github.io/rfcs/2145-type-privacy.html#lint-3-voldemort-types-its-reachable-but-i-cant-name-it)
```

Comment:
I would personally refer to it as a "sealed trait" because it is more commonly recognized. 😄